misc/metrics/src/identify.rs

```diff
@@ -20,39 +20,79 @@
 
 use crate::protocol_stack;
 use libp2p_identity::PeerId;
-use prometheus_client::encoding::{EncodeLabelSet, EncodeMetric, MetricEncoder};
+use once_cell::sync::Lazy;
+use prometheus_client::collector::Collector;
+use prometheus_client::encoding::EncodeLabelSet;
 use prometheus_client::metrics::counter::Counter;
-use prometheus_client::metrics::family::Family;
-use prometheus_client::metrics::histogram::{exponential_buckets, Histogram};
-use prometheus_client::metrics::MetricType;
-use prometheus_client::registry::Registry;
+use prometheus_client::metrics::gauge::ConstGauge;
+use prometheus_client::registry::{Descriptor, LocalMetric, Registry};
+use prometheus_client::MaybeOwned;
+use std::borrow::Cow;
+use std::cell::RefCell;
 use std::collections::HashMap;
-use std::iter;
 use std::sync::{Arc, Mutex};
 
+static PROTOCOLS_DESCRIPTOR: Lazy<Descriptor> = Lazy::new(|| {
+    Descriptor::new(
+        "remote_protocols",
+        "Number of connected nodes supporting a specific protocol, with \
+         \"unrecognized\" for each peer supporting one or more unrecognized \
+         protocols",
+        None,
+        None,
+        vec![],
+    )
```

Comment:
Just an idea, could we make `new` a `const fn`?

Reply:
Oh, I have not thought about that. Only blocker documented in https://github.com/prometheus/client_rust/issues/139.